govcd/vdc_group_test.go

```diff
@@ -0,0 +1,414 @@
+	check.Assert(err, IsNil)
+	check.Assert(disabledVdcGroup, NotNil)
+	check.Assert(disabledVdcGroup.VdcGroup.DfwEnabled, Equals, false)
+
+}
+
+func (vcd *TestVCD) Test_GetVdcGroupByName_ValidatesSymbolsInName(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+	if vcd.skipAdminTests {
+		check.Skip(fmt.Sprintf(TestRequiresSysAdminPrivileges, check.TestName()))
+	}
+	if vcd.config.VCD.Nsxt.Vdc == "" {
+		check.Skip("Missing NSX-T config: No NSX-T VDC specified")
+	}
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointVdcGroups)
+
+	adminOrg, err := vcd.client.GetAdminOrgByName(vcd.org.Org.Name)
+	check.Assert(err, IsNil)
+	check.Assert(adminOrg, NotNil)
+	test_GetVdcGroupByName_ValidatesSymbolsInName(check, adminOrg, vcd.nsxtVdc.vdcId())
+}
+
+func test_GetVdcGroupByName_ValidatesSymbolsInName(check *C, adminOrg *AdminOrg, vdcId string) {
+	// When alias contains commas, semicolons, stars, or plus signs, the encoding may reject by the API when we try to Query it
+	// Also, spaces present their own issues
+	for _, symbol := range []string{";", ",", "+", " ", "*"} {
+
+		name := fmt.Sprintf("Test%sVdcGroup", symbol)
+
+		createdVdcGroup, err := adminOrg.CreateNsxtVdcGroup(name, "", vdcId, []string{vdcId})
+		check.Assert(err, IsNil)
+		check.Assert(createdVdcGroup, NotNil)
+
+		openApiEndpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcGroups + createdVdcGroup.VdcGroup.Id
+		PrependToCleanupListOpenApi(createdVdcGroup.VdcGroup.Name, check.TestName(), openApiEndpoint)
+
+		check.Assert(createdVdcGroup, NotNil)
+		check.Assert(createdVdcGroup.VdcGroup.Id, Not(Equals), "")
+		check.Assert(createdVdcGroup.VdcGroup.Name, Equals, name)
+		check.Assert(len(createdVdcGroup.VdcGroup.ParticipatingOrgVdcs), Equals, 1)
+
+		foundVdcGroup, err := adminOrg.GetVdcGroupByName(name)
+		check.Assert(err, IsNil)
+		check.Assert(foundVdcGroup, NotNil)
+		check.Assert(foundVdcGroup.VdcGroup.Name, Equals, name)
+
+		err = foundVdcGroup.Delete()
+		check.Assert(err, IsNil)
+	}
+}
+
+// Test_NsxtVdcGroupWithOrgAdmin additionally tests Test_CreateVdcGroup,  Test_GetVdcGroupByName_ValidatesSymbolsInName
+// and Test_NsxtVdcGroup using an org amin user with added rights which allows working with VDC groups.
+func (vcd *TestVCD) Test_NsxtVdcGroupWithOrgAdmin(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+	if vcd.skipAdminTests {
+		check.Skip(fmt.Sprintf(TestRequiresSysAdminPrivileges, check.TestName()))
+	}
+	if vcd.config.VCD.Nsxt.Vdc == "" {
+		check.Skip("Missing NSX-T config: No NSX-T VDC specified")
+	}
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointVdcGroups)
+
+	adminOrg, err := vcd.client.GetAdminOrgByName(vcd.org.Org.Name)
+	check.Assert(err, IsNil)
+	check.Assert(adminOrg, NotNil)
+
+	rightsToRemove, orgAdminClient, err := newOrgAdminUserWithVdcGroupRightsConnection(check, adminOrg, "test-user", "CHANGE-ME", vcd.config.Provider.Url, true)
```

Comment:
Would be good to use different username than `test-user` because this test does not explicitly remove it until clenaup phase and there is already a user with such name being created in `catalog_test.go`